test/e2e/common/kubelet_etc_hosts.go

```diff
@@ -62,58 +62,39 @@ var _ = framework.KubeDescribe("KubeletManagedEtcHosts", func() {
 
 		By("Running the test")
 		config.verifyEtcHosts()
-
-		config.cleanup()
 	})
 })
 
 func (config *KubeletManagedHostConfig) verifyEtcHosts() {
 	By("Verifying /etc/hosts of container is kubelet-managed for pod with hostNetwork=false")
-	assertManagedStatus(config, etcHostsPodName, true, "busybox-1")
-	assertManagedStatus(config, etcHostsPodName, true, "busybox-2")
+	assertManagedStatus(config, etcHostsPodName, true, "busybox-1", true)
+	assertManagedStatus(config, etcHostsPodName, true, "busybox-2", true)
 
-	By("Verifying /etc/hosts of container is not kubelet-managed since container specifies /etc/hosts mount")
-	assertManagedStatus(config, etcHostsPodName, false, "busybox-3")
+	// Check if local /etc/hosts has a managed by kubernetes header
+	headerPresent := false
+	if data, err := ioutil.ReadFile("/etc/hosts"); err == nil {
```

Comment:
this only works if the test is run on the same machine as the one containing this file, right? it seems like a better test would be to mount /etc/hosts in two locations in the pod (e.g. /etc/hosts and /etc-hosts-test) and make sure they were equal as seen by the pod (e.g. nothing special was done to the /etc/hosts mount)